provider/pkg/gen/schema.go

```diff
@@ -907,10 +907,14 @@ func (g *packageGenerator) genProperties(typeName string, typeSchema *discovery.
 				result.requiredSpecs.Add(sdkName)
 			}
 
-			result.specs[sdkName] = schema.PropertySpec{
+			p := schema.PropertySpec{
 				Description: clearDescription(prop.Description),
 				TypeSpec:    *typeSpec,
 			}
+			if isDeprecated(prop.Description) {
+				p.DeprecationMessage = prop.Description
```

Comment:
It's not ideal to copy the entire description here, but I don't think that the deprecation notices are in a standard format. This is probably good enough for now, but we might add a TODO to parse the deprecation notice heuristically.